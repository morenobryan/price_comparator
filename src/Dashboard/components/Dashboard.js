import React from 'react';
import { StyleSheet, Text, View } from 'react-native';
import NavButton from './NavButton';

const bagIcon = '\uf100';
const toiletPaper1 = '\uf101';
const toiletPaper = '\uf102';
const drink = '\uf103';
const water = '\uf104';

export default ({ navigation }) => (
  <View style={styles.container}>
    <Text style={styles.title}>Escolha a categoria</Text>
    <View style={styles.buttonContainer}>
      <NavButton
        backgroundColor="#1A237E"
        color="#C5CAE9"
        iconColor="#C5CAE9"
        navigation={navigation}
        icon={drink}
        text="LÍQUIDOS"
        pageName="ComparePaper"
      />
      <NavButton
        backgroundColor="#1A237E"
        color="#C5CAE9"
        iconColor="#C5CAE9"
        navigation={navigation}
        icon={bagIcon}
        text="SÓLIDOS"
        pageName="ComparePaper"
      />
      <NavButton
        backgroundColor="#1A237E"
        color="#C5CAE9"
        iconColor="#C5CAE9"
        navigation={navigation}
        icon={toiletPaper}
        text="PAPEL HIGIÊNICO"
        pageName="ComparePaper"
      />
    </View>
  </View>
);

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#3949AB',
    paddingTop: 50,
    paddingBottom: 30,
  },
  title: {
    fontFamily: 'proximaNovaAltBold',
    fontSize: 30,
    marginBottom: 30,
    alignSelf: 'center',
    color: '#7986CB',
  },
  buttonContainer: {
    flex: 1,
    flexDirection: 'column',
    justifyContent: 'center',
    alignItems: 'stretch',
  },
});
