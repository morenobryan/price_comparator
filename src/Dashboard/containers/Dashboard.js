// @flow
import { connect } from "react-redux";

import {} from "../actions";
import {} from "../selectors";

import Dashboard from "../components/Dashboard";

const mapStateToProps = state => ({});

const mapDispatchToProps = dispatch => ({});

export default connect(mapStateToProps, mapDispatchToProps)(Dashboard);
