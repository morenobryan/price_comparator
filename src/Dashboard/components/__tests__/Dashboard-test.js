import 'react-native';
import React from 'react';
import Dashboard from '../Dashboard';

import renderer from 'react-test-renderer';

const props = {
  navigation: {
    navigate: () => {},
  },
};

test('renders correctly', () => {
  const tree = renderer.create(<Dashboard {...props} />).toJSON();
  expect(tree).toMatchSnapshot();
});
