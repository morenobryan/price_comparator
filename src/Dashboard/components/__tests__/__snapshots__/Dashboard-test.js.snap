// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<ExponentLinearGradient
  colors={
    Array [
      4294634711,
      4294898658,
    ]
  }
  end={undefined}
  locations={null}
  start={undefined}
  style={
    Object {
      "flex": 1,
      "paddingBottom": 30,
      "paddingTop": NaN,
    }
  }
>
  <Image
    source={1}
    style={
      Object {
        "alignSelf": "center",
        "height": 75,
        "marginBottom": 15,
        "width": 75,
      }
    }
  />
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Object {
        "alignSelf": "center",
        "color": "#3D3527",
        "fontFamily": "proximaNovaSoftBold",
        "fontSize": 30,
        "marginBottom": 30,
      }
    }
  >
    Escolha a categoria
  </Text>
  <View
    style={
      Object {
        "alignItems": "stretch",
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "center",
      }
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#FAAF08",
          "elevation": 3,
          "flexDirection": "row",
          "opacity": 1,
          "padding": 30,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Object {
            "alignSelf": "center",
            "color": "#FFF",
            "fontFamily": "flaticon",
            "fontSize": 50,
            "marginRight": 30,
          }
        }
      >
        
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Object {
            "alignSelf": "center",
            "color": "#FFF",
            "fontFamily": "proximaNovaAltRegular",
            "fontSize": 24,
          }
        }
      >
        LÍQUIDOS
      </Text>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#FA812F",
          "elevation": 3,
          "flexDirection": "row",
          "opacity": 1,
          "padding": 30,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Object {
            "alignSelf": "center",
            "color": "#FFF",
            "fontFamily": "flaticon",
            "fontSize": 50,
            "marginRight": 30,
          }
        }
      >
        
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Object {
            "alignSelf": "center",
            "color": "#FFF",
            "fontFamily": "proximaNovaAltRegular",
            "fontSize": 24,
          }
        }
      >
        SÓLIDOS
      </Text>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#FA4032",
          "elevation": 3,
          "flexDirection": "row",
          "opacity": 1,
          "padding": 30,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Object {
            "alignSelf": "center",
            "color": "#FFF",
            "fontFamily": "flaticon",
            "fontSize": 50,
            "marginRight": 30,
          }
        }
      >
        
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Object {
            "alignSelf": "center",
            "color": "#FFF",
            "fontFamily": "proximaNovaAltRegular",
            "fontSize": 24,
          }
        }
      >
        PAPEL HIGIÊNICO
      </Text>
    </View>
  </View>
</ExponentLinearGradient>
`;
